src/sdam/description/server.rs

```diff
@@ -16,6 +16,7 @@ const DRIVER_MIN_WIRE_VERSION: i32 = 6;
 const DRIVER_MAX_WIRE_VERSION: i32 = 7;
 
 #[derive(Debug, Clone, Copy, Eq, PartialEq)]
+#[non_exhaustive]
 pub enum ServerType {
```

Comment:
this one might be exhaustive. I imagine if the drivers or server introduced a new server type, it would break a lot of things.

Reply:
I wasn't sure about additional replica set types; there are already so many that I wouldn't be surprised if a new type was added. This is also only exposed because of the selection criteria predicate; I don't think it's too much of a burden to require predicates to be non-exhaustive with their matching of the server types.

Reply:
Leaving as-is per offline discussion